src/GNDStk/Node/src/one.hpp

```diff
@@ -0,0 +1,62 @@
+
+// -----------------------------------------------------------------------------
+// Node::one(string)
+//
+// Searches the node's children for a child node with the given name. If found,
+// returns the child node in question. Else, returns an empty node. Call when
+// you expect exactly one such value. (Many GNDS child nodes are expected just
+// once, even though the source file format, e.g. XML, might allow any number.)
+// With one(), as opposed to all(), you get back a direct reference to the one
+// found node, not to a container of nodes.
+// -----------------------------------------------------------------------------
+
+// const
+const Node &one(
+   const std::string &key,
+   bool &found = detail::default_bool
+) const {
+   // ""
+   // special meaning: return current node
+   if (key == "")
+      return found = true, *this;
+
+   // search
+   const Node *theone = nullptr;
+   unsigned count = 0;
+   for (auto &c : children)
+      if (c->name == key && count++ == 0)
+         theone = &(*c);
+
+   // found
+   if (count) {
+      if (count > 1)
+         log::warning(
+            "Node::one(\"{}\") called, but {} such child nodes were found",
+            key, count
```

Comment:
Do we need to add a message stating that you can't predict which child is returned in this situation?

Reply:
That's probably a good idea. At some point it would be good to review, in one fell swoop, all of the diagnostics that GNDStk can generate, and give absolutely all of them a consistent look and feel. I think they're adequate right now, but would agree that they could be better. A larger discussion, at some point, would be how much (or how little) we should do in terms of catching exceptions and, say, printing contextual information as the call stack unwinds. Right now, I'm doing that in some places, but not absolutely everywhere in which further code could possibly throw. A purist might argue that printing *any* context is really the job of an admin code - say, a debugger. Maybe, and that would make our job easier. Then again, admin codes can, like compilers, produce spectacularly convoluted and Byzantine diagnostics, whereas a human like you or I can design messages that actually make sense. So, I think we'll want to bookmark this as another for-the-future substantial discussion.